src/main/java/org/cirdles/topsoil/app/TSVTable.java

```diff
@@ -42,15 +55,96 @@
+                                deselectedRows.add(tableRow);
+                            } else {
+                                deselectedRows.remove(tableRow);
+                            }
+                        });      
+                this.getSelectionModel().clearSelection();
+            });                
+
+            //Add context menu
+            rowMenu.getItems().addAll(deselectItem);
+            
+            // only display context menu for non-empty rows
+            row.contextMenuProperty().bind(
+                    Bindings.when(Bindings.isNotNull(row.itemProperty()))
+                            .then(rowMenu)
+                            .otherwise((ContextMenu) null));
+                                 
+            return row;
+        });        
+        
+        //---------- Trying to get the sort event -> not working ---------------
+        this.getItems().addListener(new ListChangeListener<Entry>() {
+            @Override
+            public void onChanged(ListChangeListener.Change<? extends Entry> c) {
+                System.out.println("Changed");
+            }
+        });
+        
+        this.sortPolicyProperty().addListener(new ChangeListener() {
+            @Override
+            public void changed(ObservableValue observable, Object oldValue, Object newValue) {
+                System.out.println("Changed");
+            }
+        });
+        //----------------------------------------------------------------------
+        
+        deselectedRows.addListener(new ListChangeListener() {
+            @Override
+            public void onChanged(ListChangeListener.Change c) {
+                 while (c.next()) {
+                     if(c.wasRemoved()) {
+                         c.getRemoved().stream().forEach(new Consumer<TableRow<Entry>>() {
+                             @Override
+                             public void accept(TableRow<Entry> tableRow) {
+                                 tableRow.setOpacity(1);
+                                 tableRow.getItem().setIsSelected(true);
+                                 tableRow.getContextMenu().getItems().get(0).setText("Deselect");
+                             }
+                         });
+                     } else if(c.wasAdded()) {
+                         c.getAddedSubList().stream().forEach(new Consumer<TableRow<Entry>>() {
+                             @Override
+                             public void accept(TableRow<Entry> tableRow) {
+                                 tableRow.setOpacity(0.35);
+                                 tableRow.getItem().setIsSelected(false);
+                                 tableRow.getContextMenu().getItems().get(0).setText("Select");
+                             }
+                         });
+                     }
+                 }
+            };
+        });
```

Comment:
This is also too large. Please make this easier to read with a new class or something. Keep in mind that you can also write these as methods and then do `deselectedRows.addListener(this::theMethodYouMade);`.